frontend/packages/console-plugin-sdk/src/typings/nav-items.ts

```diff
@@ -87,10 +88,5 @@ export const isSeparatorNavItem = (e: Extension): e is SeparatorNavItem => {
 };
 
 export const isNavItem = (e: Extension): e is NavItem => {
-  return (
-    isHrefNavItem(e) ||
-    isResourceNSNavItem(e) ||
-    isResourceClusterNavItem(e) ||
-    isSeparatorNavItem(e)
```

Comment:
Why are you removing this check? `isNavItem` is supposed to include all nav item extensions.

I do see an issue in `perspective-nav.tsx` though:

```
        if (isNavItem(item)) {
          return createLink(item, true);
        }
        if (isSeparatorNavItem(item)) {
          return <NavItemSeparator key={`separator-${index}`} />;
        }
```

Should be:
```
        if (isSeparatorNavItem(item)) {
          return <NavItemSeparator key={`separator-${index}`} />;
        }
        return createLink(item, true);
```

Reply:
We can keep this as it was before. I just take this from the origin request but don't see a difference.

Reply:
Applied your recommendation and rebased everything.